sideboard/config.py

```diff
@@ -38,36 +99,64 @@ def get_dirnames(pyname, plugin):
     package under the configured "plugins_dir" directory.
 
     These two directories are also automatically inserted into plugin config
-    files as "root" and "module_root" and are available for interpolation.  For
-    example, a plugin could have a line in their config file like
+    files as "root" and "module_root" and are available for interpolation. For
+    example, a plugin could have a line in their config file like::
+
         template_dir = "%(module_root)s/templates"
+
+
     and that would be interpolated to the correct absolute path.
 
-    This function takes two parameters:
-    pyname: the __file__ of the module needing to know these directories
-    plugin: boolean indicating whether this is a plugin or Sideboard itself
+    Parameters:
+        requesting_file_path (str): The __file__ of the module requesting the
+            "root" and "module_root" directories.
+        is_plugin (bool): Indicates whether a plugin is making the request or
+            Sideboard itself is making the request.
     """
-    module_dir = os.path.dirname(os.path.abspath(pyname))
-    if plugin:
+    module_dir = os.path.dirname(os.path.abspath(requesting_file_path))
+    if is_plugin:
         from sideboard.lib import config
         plugin_name = os.path.basename(module_dir)
-        root_dir = os.path.join(config['plugins_dir'], plugin_name.replace('_', '-'))
+        root_dir = os.path.join(config['plugins_dir'], plugin_name)
```

Comment:
Indeed, we need to keep this replace.  Suppose we have a Sideboard plugin called ``foo-bar``.  We wanted to follow the standard .deb and .rpm packaging conventions so that when building e.g. an RPM then we would have the package be named ``foo-bar.rpm`` rather than ``foo_bar.rpm``, and thus it also looks in ``/etc/sideboard/plugins.d/foo-bar.cfg`` for its config file (which is what this line is doing).

Obviously Python modules need to use underscores, which means that a lot of our packaging would do things like drop our main module file into ``/opt/sideboard/plugins/foo-bar/foo_bar/__init__.py`` which seems a little silly but is how we've always done it.

MAGFest doesn't care about any of this, because we run everything out of cloned Git repos so we're not doing any actual packaging and don't look in ``/etc/sideboard`` for any config files whatsoever.  But it's important for projects which do care about packaging, e.g. my day job :)